src/drivers/actuators/vertiq_io/vertiq_io.cpp

```diff
@@ -0,0 +1,317 @@
+		_serial_interface.InitSerial(_telemetry_device, _param_vertiq_baud.get());
+		_request_telemetry_init.store(false);
+	}
+
+	//Handle IQUART reception and transmission
+	_client_manager.HandleClientCommunication();
+
+	// If we're supposed to ask for telemetry from someone
+	if (_telem_bitmask) {
+		//Grab the next ID, and if it's real get it ready to send out
+		uint16_t next_telem = _telem_manager.UpdateTelemetry();
+
+		if (next_telem != _impossible_module_id) {
+			_transmission_message.telem_byte = next_telem;
+		}
+	}
+
+	//Get the most up to date version of our parameters/check if they've changed
+	parameters_update();
+
+	//Make sure we also update the mixing output to get the most up to date configuration
+	_mixing_output.update();
+	_mixing_output.updateSubscriptions(true);
+
+	//Go ahead and check to see if our actuator test has gotten anything new
+	if (_actuator_test_sub.updated()) {
+		_actuator_test_sub.copy(&_actuator_test);
+
+		//Our test is active if anyone is giving us commands through the actuator test
+		_actuator_test_active = _actuator_test.action == actuator_test_s::ACTION_DO_CONTROL;
+	}
+
+	//stop our timer
+	perf_end(_loop_perf);
+}
+
+
+void VertiqIo::parameters_update()
+{
+	//If someone has changed any parameter in our module. Checked at 1Hz
+	if (_parameter_update_sub.updated()) {
+		//Grab the changed parameter with copy (which lowers the "changed" flag)
+		parameter_update_s param_update;
+		_parameter_update_sub.copy(&param_update);
+
+		// If any parameter updated, call updateParams() to check if
+		// this class attributes need updating (and do so).
+		updateParams();
+
+		//If the target module ID changed, we need to update all of our parameters
+		if (_param_vertiq_target_module_id.get() != _configuration_handler.GetObjectIdNow()) {
+			_configuration_handler.UpdateClientsToNewObjId(_param_vertiq_target_module_id.get());
+
+			//Make sure we start a new read!
+			_param_vertiq_trigger_read.set(true);
+			_param_vertiq_trigger_read.commit_no_notification();
+		}
+
+		//If you're set to re-read from the motor, mark all of the IQUART parameters for reinitialization, reset the trigger, and then update the IFCI params
+		if (_param_vertiq_trigger_read.get()) {
+			_configuration_handler.MarkConfigurationEntriesForRefresh();
+			_param_vertiq_trigger_read.set(false);
+			_param_vertiq_trigger_read.commit_no_notification();
+		}
+
+		_configuration_handler.UpdateIquartConfigParams();
+	}
+}
+
+void VertiqIo::OutputControls(uint16_t outputs[MAX_ACTUATORS]){
+	//Put the mixer outputs into the output message
+	for(uint8_t i = 0; i < _transmission_message.num_cvs; i++){
+		_transmission_message.commands[i] = outputs[i];
+	}
+
+	_operational_ifci.PackageIfciCommandsForTransmission(&_transmission_message, _output_message, &_output_len);
+	_operational_ifci.packed_command_.set(*_serial_interface.GetIquartInterface(), _output_message, _output_len);
+	_serial_interface.ProcessSerialTx();
```

Comment:
Similar to above in the configuration handler, might it make sense to ask the Client Manager to HandleClientCommunication here instead of directly calling on the serial interface? So you are always letting the manager handle the client communication everywhere? 

Or you could even just not do any sending here and let the standard HandleClientCommunication on the next run through pick it up, unless you are concerned about the buffer running out of space

Reply:
Yea, I'd be a bit concerned about the buffer filling up. The updateOutputs gets called at the mixer frequency, not the run loop. Just for extra security with throttle commands, I want the explicit process tx to make sure we get our controls out